src/core/mac/mac_frame.cpp

```diff
@@ -54,16 +54,14 @@ void HeaderIe::Init(uint16_t aId, uint8_t aLen)
     SetLength(aLen);
 }
 
-void Frame::InitMacHeader(Type             aType,
-                          Version          aVersion,
-                          const Addresses &aAddrs,
-                          const PanIds    &aPanIds,
-                          SecurityLevel    aSecurityLevel,
-                          KeyIdMode        aKeyIdMode,
-                          bool             aSuppressSequence)
+uint16_t Frame::InitFcf(Type             aType,
```

Comment:
Rename to `DetermineFcf()`?
```suggestion
uint16_t Frame::DetermineFcf(Type             aType,
```

Reply:
My point is that this method doesn't actually "initialize" the frame; it calculates and returns the FCF value based on a set of inputs. So, I suggest using the name `DetermineFcf()` instead of `InitFcf()`.

Also, can consider defining this method as `static`?

Reply:
Updated